Gjoranv/create registry from config

gjoranv opened the pull request:
@bratseth, please review the new constructor code carefully.

Next, we can have this injected to SearchHandler instead of creating a new registry there.

gjoranv committed:
Allow creating a CompiledQueryProfileRegistry from config.

gjoranv committed:
Add CompiledQueryProfileRegistry component for injection.

gjoranv committed:
Follow naming convention by using singular 'profileRegistry'

baldersheim reviewed (changes requested):
Looks good to me. I would just include the change in SearchHandler to avoid blink-dpa being the first one using it.

gjoranv committed:
Make the main constructor take a CompiledQueryProfileRegistry

.. instead of QueryProfilesConfig

gjoranv committed:
Add the query profile registry to the set of test components

.. for all integration style tests.

Note that these tests were written before the DI framework, and
typically used the former 'handlers' config as only handlers
and searchers could be configured at the time.

gjoranv commented:
Done, but that was a _painful_ reminder about the integration style tests in container-search. These were written before the DI framework, but were adapted to it by converting the test files for configs that are long gone now (handlers and chains) into the new `components` config.

IMO, these tests should be rewritten to plain unit tests, and some should just be removed. No need to test DI here anymore.

baldersheim reviewed (approved):
LGTM. Sometimes pain is necessary go get going :)

bratseth commented on container-search/src/main/java/com/yahoo/search/query/profile/compiled/CompiledQueryProfileRegistry.java:
```diff
@@ -18,6 +23,15 @@
+    public CompiledQueryProfileRegistry(QueryProfilesConfig config) {
+        QueryProfileRegistry registry = QueryProfileConfigurer.createFromConfig(config);
+        typeRegistry = registry.getTypeRegistry();
+        for (QueryProfile inputProfile : registry.allComponents()) {
```
Why not just do 
QueryProfileCompiler.compile(registry)
here?

gjoranv replied:
I didn't want to create a temporary object of the same type as the class itself. There is room for improvement, but it requires creating even more public methods and/or constructors.

bratseth reviewed (approved):
LGTM but what is it that you want to achieve, recompiling less?
